uuid/uuid_test.go

```diff
@@ -0,0 +1,74 @@
+package uuid_test
+
+import (
+	"crypto/md5"
+	"testing"
+
+	gofrsuuid "github.com/gofrs/uuid"
+	gluuid "github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-go-kit/uuid"
+)
+
+var (
+	uuidGOOGLE gluuid.UUID
+	uuidGOFRS  gofrsuuid.UUID
+)
+
+func init() {
+	gluuid.EnableRandPool()
+}
+
+func FuzzGetMD5UUID(f *testing.F) {
+	f.Add("hello")
+	f.Add("")
+	f.Add(gluuid.New().String())
+
+	f.Fuzz(func(t *testing.T, data string) {
+		googleMd5, err := uuid.GetMD5UUID(data)
+		require.NoError(t, err)
+
+		md5Sum := md5.Sum([]byte(data))
+		gofrsMd5, err := gofrsuuid.FromBytes(md5Sum[:])
+		require.NoError(t, err)
+		gofrsMd5.SetVersion(gofrsuuid.V4)
+		gofrsMd5.SetVariant(gofrsuuid.VariantRFC4122)
+
+		require.Equal(t, gofrsMd5.String(), googleMd5.String())
+	})
+}
+
+func Test_fastUUID(t *testing.T) {
+	t.Run("test google conversion gofrs", func(t *testing.T) {
+		uuidGOOGLE = gluuid.New()
+		b, _ := uuidGOOGLE.MarshalBinary()
+		uuidGOFRS = gofrsuuid.FromBytesOrNil(b)
+		require.Equal(t, uuidGOOGLE.String(), uuidGOFRS.String())
+	})
+}
+
+func Benchmark_GOOGLE_UUID(b *testing.B) {
```

Comment:
If we move all these benchmarks inside a single function and have them a sub-benchmarks is going to be much easier to fire them from an IDE and compare results.